payu/models/cice.py

```diff
@@ -249,13 +283,11 @@ def setup(self):
         else:
             run_runtime = setup_nml['npt']*setup_nml['dt']
 
-        # Now write out new run start date and total runtime.
+        # Add the prior runtime and new runtime to the working copy of the
+        # CICE namelist.
         setup_nml['npt'] = run_runtime / setup_nml['dt']
-        assert(total_runtime % setup_nml['dt'] == 0)
-        setup_nml['istep0'] = int(total_runtime / setup_nml['dt'])
-
-        nml_path = os.path.join(self.work_path, self.ice_nml_fname)
-        self.ice_in.write(nml_path, force=True)
+        assert (prior_runtime_seconds % setup_nml['dt'] == 0)
```

Comment:
Good question, I'm not sure what its checking for either. I think you're right that the only way it could fail is if `dt` was changed, but then it wouldn't even reliably fail depending on the new value.

Do you think it's worth guarding against changes to `dt` at all? E.g. would it ever be valid for someone to try and set up ESM with a different ocean/ice timestep?

Reply:
Actually I think it is valid to change the timestep, it might get used if someone has messed with some forcing / boundary condition and then had issues with the model being unstable. Reducing the timestep can help get past instabilities.

Reply:
Needs a different PR 😅. Well just to check if it would work in reality

Reply:
Good point! So maybe we just keep the existing behaviour for now and work out whether to change it in a separate issue

Reply:
Yeah I think so, it would be nice to get rid of the `assert` but also it probably doesn't break anything so lets leave it there